### PR TITLE
Fix transfer timeout for HTTP/2

### LINKED ISSUE #260: Body transfer timeout possibly not enforced correctly
More details will follow.

### PR DESCRIPTION
**Pull request by kelunik**
Fixes #260.

### CODE REVIEW

File: src/Connection/Internal/Http2ConnectionProcessor.php [hunk 1]
```diff
@@ -905,29 +918,34 @@ public function request(Request $request, CancellationToken $cancellationToken,
...
             );
 
-            if ($request->getTransferTimeout() > 0) {
-                // Cancellation token combined with timeout token should not be stored in $stream->cancellationToken,
-                // otherwise the timeout applies to the body transfer and pushes.
-                $cancellationToken = new CombinedCancellationToken(
-                    $cancellationToken,
-                    new TimeoutCancellationToken($request->getTransferTimeout())
-                );
-            }
```
Comment (kelunik) [inline src/Connection/Internal/Http2ConnectionProcessor.php]: Seems like this commend has been introduced by you, @trowski: 93c2e6e5baab68558deb8c0c3e7b8131726ba1db

Comment (trowski) [reply src/Connection/Internal/Http2ConnectionProcessor.php]: I think my assumption at that time was that the transfer timeout only applied to the headers and that there was a separate timeout for transferring the response body.

**Comment by kelunik** [comment]
@trowski I'm not sure how pushes should be handled if the parent is cancelled / times out. Should there be a difference in handling? I guess for cancelling, cancelling the pushes makes sense, because they're essentially children of the parent stream.

**Comment by trowski** [comment]
I agree that a push is essentially a child of the parent request. If the parent is cancelled for any reason (timeout, etc.), all the pushes associated should be cancelled.

**Pull request merged**